### ISSUE TITLE
Loss change depending on the number of epochs chosen?

### ISSUE THREAD

**Issue by PabloRR100** [keras-team/keras#8739]
Hi colleagues, 

I am using Keras to train different NNs. I would like to know why if I increment the epochs, the result until the same number of epochs is not the same for the evolution of loss. I am using shuffle=False, and np.random.seed(2017), tf.set_random_seed(2017), and I have check that if I repeat with the same number of epochs, the result is the same, so the random initialization is working. After the epochs training, I am deleting the ANN so the training begins from 0 again.

Here I attach the picture of the resulting training with 10 epochs:

<img width="965" alt="captura de pantalla 2017-12-09 a las 8 58 07" src="https://user-images.githubusercontent.com/23745991/33793851-fac96fdc-dcbf-11e7-916e-5b496d4e1524.png">

And here I attach the picture of the resulting training with 8 epochs:

<img width="962" alt="captura de pantalla 2017-12-09 a las 8 59 04" src="https://user-images.githubusercontent.com/23745991/33793855-06b43372-dcc0-11e7-8082-e7f6c4a87dc1.png">

Also, I would like to know why the training time is not exactly (8/10) the 10 epochs attempt and how is it possible that some of them have less accuracy with 2 more epochs!

Here is the link to open code Jupyter Notebook. [ GitHub Jupyter Notebook - ANN Comparison](https://github.com/PabloRR100/Pruning-Algorithm-Method/blob/master/NN-Comparison.ipynb)

Thanks a lot!
